### PR TITLE
Code Enhancements & README

### PR DESCRIPTION
**Pull request by pauljohanneskraft**
# *Name of the PR*

## :recycle: Current situation
*Describe the current situation (if possible with and exemplary (or real) code snippet and/or where this is used)*

## :bulb: Proposed solution
*Describe the solution (if possible with and exemplary (or real) code snippet)*

### Problem that is solved
*Provide a description and link issues that are solved*

### Implications
*Describe the implications, e.g. refactoring*

## :heavy_plus_sign: Additional Information
*Provide some additional information if possible*

### Related PRs
*Reference the related PRs*

### Testing
*Are there tests included? If yes, which situations are tested and which corner cases are missing?*

### Reviewer Nudging
*Where should the reviewer start, where is a good entry point?*

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,18 +1,112 @@
...
-We provide a small example proejct to showcase Presenter.
-Clone the repo, navigate in the root folder and start the example application using `swift run Example`.
+We provide a small example project to showcase Presenter.
+
+Clone the repo, navigate in the root folder and start the example server application using `swift run Example`.
+To check out how to use Presenter on the client side, see the [example client application](Example).
 
 ## Contributing
+
 Contributions to this projects are welcome. Please make sure to read the [contribution guidelines](https://github.com/Apodini/.github/blob/release/CONTRIBUTING.md) first.
```
Comment (PSchmiedmayer) [inline README.md:108]:
```suggestion
Contributions to this project are welcome. Please make sure to read the [contribution guidelines](https://github.com/Apodini/.github/blob/main/CONTRIBUTING.md) and the [contributor covenant code of conduct](https://github.com/Apodini/.github/blob/main/CODE_OF_CONDUCT.md) first.
```

File: README.md [hunk 2]
```diff
@@ -1,18 +1,112 @@
...
+        .padding(16)
+        .background(Color(red: 1, green: 1, blue: 1))
+        .shadow(radius: 8)
+        .padding(16)
+    }
+
+}
+```
+
+This example above already showcases some of the more complicated features of SwiftUI, including state management and local actions.
```
Comment (PSchmiedmayer) [inline README.md:67]: Would be nice to show some example code how the View is then transitioned into a data representation just to make this example complete and e.g. show how to define a view, instantiate it and then decode it into a data representation. How this data is then returned (e.g. in a Vapor or Apodini) server is then out of the scope of the README.

File: README.md [hunk 3]
```diff
@@ -1,18 +1,112 @@
...
+                Text("Error: " + error.localizedDescription)
+            }
+        }
+    }
+
+}
+```
+
+When using plugins, custom views, custom view modifiers or actions, please make sure to register them during app start using  `Presenter.use(plugin:)`, `Presenter.use(view:)`, `Presenter.use(modifier:)` or `Presenter.use(action:)` respectively. Tip: You can also create an app-specific plugin to register multiple views, view modifiers and actions at once.
+
```
Comment (PSchmiedmayer) [inline README.md:98]: Would be nice to show how to interact with native client code and/or use plugins here but maybe this is then part of a separate .md file somewhere in a Documentation folder and a next time.

File: README.md [hunk 4]
```diff
@@ -1,18 +1,112 @@
 # Presenter
 
-## Requirements
```
Comment (PSchmiedmayer) [inline README.md:3]: Would be nice to also show some badges here similar to https://github.com/Apodini/ApodiniCollector
You can use the following badge for the DOI:
[DOI](https://zenodo.org/badge/latestdoi/377776151)

**Review by PSchmiedmayer — CHANGES_REQUESTED** [review]
I took a look at the PR and I like the new additions 👍
I had a few comments in the READMe and have a few general comments mostly related to the example:
- You didn't commit the Xcode projects in the repo as they are part of the .gitignore file
- You can simplify the structure of the Example app and do not need asset catalogues or info.plist with the new Xcode 13 project format. You can take a look at example in the https://github.com/Apodini/CollectorAnalystPresenterExample repo and can also take inspiration from the setup and the CI testing the client application over there
- I would move the example web service in the example folder and create a Xcode Workspace to be able to open the server and client apps at the same time. Also similar to the setup found in the https://github.com/Apodini/CollectorAnalystPresenterExample repo.

**Comment by PSchmiedmayer** [comment]
@pauljohanneskraft Is there an update on this PR and fixing the issues when getting the presenter views of the system in the [CollectorAnalystPresenterExample](https://github.com/Apodini/CollectorAnalystPresenterExample) repo?
